File: backend/app/schemas/document.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum

class DocumentCategory(str, Enum):
    """Categorias de documentos do condomínio"""
    REGIMENTOS = "regimentos"
    ATAS = "atas"
    COMUNICADOS = "comunicados"
    RELATORIOS = "relatorios"
    OUTROS = "outros"

class FileUploadResult(BaseModel):
    """Resultado do upload de um arquivo individual"""
    filename: str
    status: str  # 'success' | 'failed'
    document_id: Optional[str] = None
    error: Optional[str] = None
    message: str

class DocumentUploadResponse(BaseModel):
    """Resposta do upload de múltiplos arquivos"""
    total_files: int
    successful: int
    failed: int
    results: List[FileUploadResult]


class DocumentResponse(BaseModel):
    id: str
    filename: str
    file_type: str
    file_size: int
    upload_date: datetime
    status: str
    category: DocumentCategory
    tenant_id: str

    class Config:
        from_attributes = True


class DocumentListResponse(BaseModel):
    documents: List[DocumentResponse]
    total: int


class ChatRequest(BaseModel):
    question: str
    max_chunks: int = 5  # Número de chunks a recuperar


class ChatResponse(BaseModel):
    answer: str
    sources: List[dict]  # Lista de documentos citados
    confidence: Optional[float] = None

File: frontend/admin/src/types/document.ts
/**
 * Types e interfaces para sistema de documentos
 */

export type DocumentCategory =
    | 'regimentos'
    | 'atas'
    | 'comunicados'
    | 'relatorios'
    | 'outros'

export type DocumentStatus =
    | 'uploading'
    | 'processing'
    | 'extracting'
    | 'chunking'
    | 'embedding'
    | 'completed'
    | 'failed'

export interface Document {
    id: string
    filename: string
    file_type: string
    file_size: number
    upload_date: string
    status: DocumentStatus
    category: DocumentCategory
    tenant_id: string
}

export interface FileUploadResult {
    filename: string
    status: 'success' | 'failed'
    document_id?: string
    error?: string
    message: string
}

export interface DocumentUploadResponse {
    total_files: number
    successful: number
    failed: number
    results: FileUploadResult[]
}

export interface DocumentListResponse {
    documents: Document[]
    total: number
}

/**
 * Labels amigáveis para categorias de documentos
 */
export const CATEGORY_LABELS: Record<DocumentCategory, string> = {
    regimentos: 'Regimentos',
    atas: 'Atas',
    comunicados: 'Comunicados',
    relatorios: 'Relatórios',
    outros: 'Outros'
}

/**
 * Labels amigáveis para status de processamento
 */
export const STATUS_LABELS: Record<DocumentStatus, string> = {
    uploading: 'Enviando',
    processing: 'Processando',
    extracting: 'Extraindo texto',
    chunking: 'Fragmentando',
    embedding: 'Gerando embeddings',
    completed: 'Concluído',
    failed: 'Falhou'
}

/**
 * Cores para cada status
 */
export const STATUS_COLORS: Record<DocumentStatus, string> = {
    uploading: 'text-blue-400',
    processing: 'text-yellow-400',
    extracting: 'text-yellow-400',
    chunking: 'text-yellow-400',
    embedding: 'text-yellow-400',
    completed: 'text-green-400',
    failed: 'text-red-400'
}
